src/image-service.ts

```diff
@@ -43,18 +49,49 @@ export const ImageService = {
     }
 
     const auth = {} // Images are public so not authentication info is required
-    await this.fetchImage(imageName, auth, docker)
+    await this.fetchImageWithRetry(imageName, auth, docker)
   },
 
-  /* Retrieve the imageName using the auth details provided, if any */
-  async fetchImage(
+  /* Retrieve the image using the auth details provided, if any with retry and backoff */
+  async fetchImageWithRetry(
     imageName: string,
     auth = {},
     docker = new Docker()
   ): Promise<void> {
-    core.info(`Pulling image ${imageName}...`)
-    const stream = await docker.pull(imageName, {authconfig: auth})
-    await endOfStream(docker, new Readable().wrap(stream))
-    core.info(`Pulled image ${imageName}`)
+    let attempt = 0
+
+    while (attempt < MAX_RETRIES) {
+      try {
+        core.info(`Pulling image ${imageName} (attempt ${attempt + 1})...`)
+        const stream = await docker.pull(imageName, {authconfig: auth})
+        await endOfStream(docker, new Readable().wrap(stream))
+        core.info(`Pulled image ${imageName}`)
+        return // Exit on success
+      } catch (error) {
+        attempt++
+
+        if (
+          error instanceof Error &&
+          (error.message.includes('429') ||
```

Comment:
The Docker API typically returns a `429 Too Many Requests` error for abuse rate limits. To ensure proper handling, I added a condition to check for either `429` or `Too Many Requests`. This way, even if one of these terms is missing, the error can still be detected, allowing for a retry.